app/controllers/nonsipp/shares/SharesListController.scala

```diff
@@ -310,14 +311,20 @@ object SharesListController {
                   routes.SharesCheckAndUpdateController.onPageLoad(srn, index).url,
                   Message("sharesList.row.check.hiddenText", sharesMessage)
                 )
-              case _ =>
+              case _ if canRemove =>
```

Comment:
Probably there is an additional guard needed on the remove shares page to prevent the removal of the pre-populated share if the remove shares page is reached by modifying the url in the browser.